### PR TITLE
Task40 initial commit

### PR DESCRIPTION
**Pull request by IlyaBaek**

### CODE REVIEW

File: Task10_Auth/src/main/java/UsersClient.java [hunk 1]
```diff
@@ -43,9 +45,25 @@ public static ResponseWrapper createUser(UsersDto usersDto) {
         return responseWrapper;
     }
 
-    public static ResponseWrapper getUsers() {
-        HttpGet httpGet = new HttpGet(PropertiesReader.get("appURI") + PropertiesReader.get("usersURI"));
-        httpGet.addHeader("Authorization", AuthSingleton.getInstance().getReadToken());
+    public static ResponseWrapper getUsers(Integer olderThan, Integer youngerThan, UsersDto.Sex sex) {
```
Comment (EugenBorisik) [inline Task10_Auth/src/main/java/UsersClient.java:50]: I think you can avoid a lot of calls of:
`UsersClient.getUsers(null, null, null)`
in tests if you will add in client's code one more method:
`public static ResponseWrapper getUsers() {
return getUsers(null, null, null);
}`
Also, such approach grant that in future you will not need to change many places with the same code

File: Task10_Auth/src/test/java/GetUsersTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import org.junit.jupiter.api.Test;
+import reseponsesDTO.UsersDto;
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class GetUsersTest {
```
Comment (EugenBorisik) [inline Task10_Auth/src/test/java/GetUsersTest.java:12]: Common comment for this class:
1. You're using not interface as type (List vs ArrayList)
2. Tests will test nothing if there is no users in app

**Review by EugenBorisik — CHANGES_REQUESTED** [review]

File: Task10_Auth/src/main/java/UsersClient.java (hunk 1 above)
Comment (IlyaBaek) [reply Task10_Auth/src/main/java/UsersClient.java:50]: done

File: Task10_Auth/src/test/java/GetUsersTest.java (hunk 2 above)
Comment (IlyaBaek) [reply Task10_Auth/src/test/java/GetUsersTest.java:12]: 2. Added BeforeTest which create the user if it's not exist

Comment (IlyaBaek) [reply Task10_Auth/src/test/java/GetUsersTest.java:12]: 1. Chhanged from ArrayList<UserDto> to List<UserDto>

File: Task10_Auth/src/test/java/ZipCodesTests.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public void addAlreadyUsedZipCodeAndCheckThereIsNoDuplicationInUsedAndNotUsedZip
         ZipCodesClient.createZipCodeIfNotExist();
         ResponseWrapper responseGet = ZipCodesClient.getZipCodes();
         List<String> availableZipCodes = (List<String>) responseGet.getResponseBody();
-        ResponseWrapper getUsersResponse = UsersClient.getUsers();
+        ResponseWrapper getUsersResponse = UsersClient.getUsers(null, null, null);
```
Comment (EugenBorisik) [inline Task10_Auth/src/test/java/ZipCodesTests.java:80]: UsersClient.getUsers()

**Review by EugenBorisik — APPROVED** [review]

**Pull request merged**